cpp/src/binaryop/compiled/binary_ops.cu

```diff
@@ -50,8 +50,11 @@ struct scalar_as_column_view {
   return_type operator()(scalar const& s, rmm::cuda_stream_view, rmm::mr::device_memory_resource*)
   {
     auto& h_scalar_type_view = static_cast<cudf::scalar_type_t<T>&>(const_cast<scalar&>(s));
-    auto col_v =
-      column_view(s.type(), 1, h_scalar_type_view.data(), (bitmask_type const*)s.validity_data());
+    auto col_v               = column_view(s.type(),
+                             1,
+                             h_scalar_type_view.data(),
+                             (bitmask_type const*)s.validity_data(),
```

Comment:
Can we eliminate the C-style cast here? This one isn't safe to do as a `static_cast` because it's effectively `bool *` to `uint32_t *` and I want that to be super obvious.
```suggestion
                             reinterpret_cast<bitmask_type const*>(s.validity_data()),
```

Reply:
Sure. To be clear, this code predates my PR, but it's small enough that I'm OK with squeezing the change in here nonetheless.